### PR TITLE
Support cancellation of subcommands in `swift-syntax-dev-utils` using Ctrl-C

### PR DESCRIPTION
**Pull request by ahoppen**
It’s really quite unfortunate that we need to do all this handling ourselves…

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

File: SwiftSyntaxDevUtils/Sources/swift-syntax-dev-utils/SwiftSyntaxDevUtils.swift [hunk 1]
```diff
@@ -31,4 +31,18 @@ struct SwiftSyntaxDevUtils: ParsableCommand {
...
   )
+
+  public static func main() {
+    SigIntListener.registerSigIntSubprocessTerminationHandler()
+    do {
+      var command = try parseAsRoot(nil)
+      try command.run()
+    } catch {
+      if !SigIntListener.hasReceivedSigInt {
+        // No point printing an error message if the user requested the termiation
```
Comment (bnbarham) [inline SwiftSyntaxDevUtils/Sources/swift-syntax-dev-utils/SwiftSyntaxDevUtils.swift:42]:
```suggestion
        // No point printing an error message if the user requested the termination
```

**Review by bnbarham — APPROVED** [review]
... unfortunate that this isn't handled by `Process` :(

File: SwiftSyntaxDevUtils/Sources/swift-syntax-dev-utils/common/ProcessRunner.swift [hunk 2]
```diff
@@ -12,6 +12,29 @@
...
+  static var hasReceivedSigInt: Bool = false
+
+  /// Registers a `SIGINT` signal handler that forwards `SIGINT` to all
+  /// subprocesses that are registered in `runningSubprocesses`
+  static func registerSigIntSubprocessTerminationHandler() {
+    #if canImport(Darwin) || canImport(Glibc)
+    signal(SIGINT) { _ in
+      SigIntListener.hasReceivedSigInt = true
+      for process in SigIntListener.runningSubprocesses {
+        kill(process.processIdentifier, SIGINT)
```
Comment (bnbarham) [inline SwiftSyntaxDevUtils/Sources/swift-syntax-dev-utils/common/ProcessRunner.swift:31]: There's a `Process.terminate` and `Process.interrupt` you could use instead signalling directly

Comment (ahoppen) [reply SwiftSyntaxDevUtils/Sources/swift-syntax-dev-utils/common/ProcessRunner.swift:31]: Oh, nice. I didn’t know about that.

**Review by bnbarham — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test

**Review by kimdv — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**